### DISCUSSION TITLE
How to efficiently update generated lazy loading images from a reused for loop

### DISCUSSION THREAD

**Discussion by simonferndriger** [alpinejs/alpine#3499] [1. Help] [answered]
### This does not update the rendered images when the images array changes
I use a classic `<template x-for="(image, index) in images">` while the template generates a lazyloading image like this `<img :src="src" x-intersect="src = image.src">`. Now, when an image is displayed and assigned a src, it does not change anymore, even though the `images` array changes :(

### This works, but does not perform well
So what I did was to add a watcher like this in the image: `<img :src="src" x-init="$watch('images', images => { if(src) src = images[index].src })" x-intersect="src = image.src">`. While this works, it gets extremly slow after adding 100+ items to the images array (up tp 10 seconds page blocking rendering).

### Thoughts
Of course, I could leave out the lazyloading part, but this would not be optimal for mobile devices. Thank you for your help in advance!

### Question
**What would be an efficient way to achieve this?**

**Comment by ekwoka** [comment, marked as answer]
just use the `loading` property on the img tag https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img

No need for x-intersect at all.

Also, you should look into the `srcset` property as well.

JS for lazy loading has been a bad idea since longer than Alpine has been around.

**Reply by simonferndriger** [reply]
Awesome, thank you! I forgot about this 🙈
